nx/test/nx/backend_test.exs

```diff
@@ -0,0 +1,103 @@
+defmodule Nx.BackendTest do
+  use ExUnit.Case, async: true
+
+  defmodule SolveTestBackend do
+    defstruct [:state]
+    alias Nx.BinaryBackend
+
+    require Logger
+
+    def from_binary(tensor, data, _opts) do
+      %{tensor | data: %__MODULE__{state: data}}
+    end
+
+    def backend_transfer(%{data: %{state: data}} = tensor, backend, _opts) do
+      %{tensor | data: struct(backend, state: data)}
+    end
+
+    def solve(_out, a, b) do
+      Logger.info("called custom implementation")
```

Comment:
Instead of use `Logger`, send a message to self and use `assert_received`.

Reply:
Duh. I tried registering the process and sending by name. Didn't occur to me that this is all within the same process!